inputremapper/injection/mapping_handlers/combination_handler.py

```diff
@@ -98,32 +104,35 @@ def notify(
             # we are not responsible for the event
             return False
 
-        was_activated = self.is_activated()
-
         # update the state
         # The value of non-key input should have been changed to either 0 or 1 at this
         # point by other handlers.
         is_pressed = event.value == 1
+        is_released = event.value == 0
         self._pressed_keys[event.input_match_hash] = is_pressed
         # maybe this changes the activation status (triggered/not-triggered)
         is_activated = self.is_activated()
 
-        if is_activated == was_activated or is_activated == self._output_state:
+        if is_activated == self._output_active:
             # nothing changed
-            if self._output_state:
-                # combination is active, consume the event
-                return True
-            else:
-                # combination inactive, forward the event
-                return False
+            # combination is active: consume the event
+            # combination inactive: forward the event
+            if is_pressed:
+                self.remember(self._output_active, event)
+                return self._output_active
+
+            if is_released:
+                # `False` means that the event-reader will forward it.
+                return not self.should_release_event(event)
 
         if is_activated:
             # send key up events to the forwarded uinput
             self.forward_release()
             event = event.modify(value=1)
-        else:
-            if self._output_state or self.mapping.is_axis_mapping():
-                # we ignore the suppress argument for release events
+
+        if not is_activated:
```

Comment:
This change may seem silly, but I honestly think the `if not` looks more readable than the `else` here. Same with is_pressed and is_released.